### ISSUE TITLE
Density map cannot be set to opacity

### ISSUE THREAD

**Issue by xizhang0819** [arose/nglview#868]
- Question about usage? Please search your issue here first: https://github.com/arose/nglview/issues/589 and https://github.com/arose/nglview/issues/785

Hello! I am using NGLview to visualize density map and protein. One issue I met is that I cannot set opacity for some density maps. The attached file contains two maps: a density map orriginally downloaded from EMDB, which can be set to opacity; and its "save map as..." saving map by Chimera, which cannot be set to opacity. Following is the code on Jupyter Notebook.

#############################
import nglview as nv

view = nv.NGLWidget()
filename1 = 'save_test.mrc'
filename2 = 'emd_1461.map'
view = nv.show_file(filename2)
view.add_component(filename2)
view

view.component_0.update_surface(color="orange", isolevel=0.5, opacity=0.5)
#################################################

Could you help me throuth this? This is the first time I post an issue on Github. If you need more information, please feel free to ask me. Thank you very much!

Xi

- Version report
nglview: 2.7.1
ipython:7.5.1
[Test_Map.zip](https://github.com/arose/nglview/files/4119059/Test_Map.zip)

**Comment by hainm** [comment]
```
view = nv.show_file(filename2)
view.add_component(filename2)
```

hi @xizhang0819  what you mean in above code (you added the `filename2` twice).

**Comment by xizhang0819** [comment]
Hi @hainm , to test if the opacity is working or not, I import a map and change its representation to opacity with smaller isovalue (larger volume). Then I add the same map as a hard core reference subject. In fact, they are not necessary to be the same. For example, we can change it to:
#######
view = nv.show_file(filename1)
view.add_component(filename2)
########
In this case, it will used to examine if filename1 can be set to opacity or not.


since I used:
#############
view.component_0.update_surface(...)
##############
therefore, the code can change the first imported file (filename1 or filename2, changable) and examine if it can be set to opacity.

**Comment by hainm** [comment]
I think you made thing much more complicated to report. 
So basically with `emd_1461.map'`, you can set the opacity and it works correctly but you can not change the opacity for  `save_test.mrc`. Is that correct?

**Comment by xizhang0819** [comment]
Yes thats correct, and both of the map can correctly show in Chimera.

**Comment by hainm** [comment]
thanks, it sounds like a bug from [NGL](https://github.com/arose/ngl.
Unfortunately it (NGL) is not actively developed anymore. cc @arose

**Comment by xizhang0819** [comment]
Thank you. I believe I locate the bug. In density map, there are three parameters to indicate the coordinate system:
mapc, mapr, maps.
Normally, we use [mapc, mapr, maps] = [1,2,3] to stand for a right-hand xyz system. However, sometimes the density map is measured and recorded under different coordinate system such as yxz or zyx, represented as [2,1,3] and [3,2,1] respectively.
I have tested all 6 combination cases of coordinate systems. Results are as follows ("N" means cannot set to opacity):
123 N
132 Y
213 Y
231 N
312 N
321 Y

As you can see, all of the right-hand coordinate systems are failed to set to opacity while all of the left-hand coordinate systems work fine. If the bug cannot be fixed, I may manully set the density map to a left-hand coordinate system for visualization.

Thank you again!

**Comment by hainm** [comment]
thanks @xizhang0819. It takes time to have the bug fixed (or it might not even fixed :D in NGL). Please go head make adjustment for your work. cheers.
